packages/client3/components/design-system/file-upload/file-upload.tsx

```diff
@@ -1,50 +1,120 @@
 import clsx from "clsx";
-import React, { forwardRef } from "react";
+import React, { forwardRef, useCallback, useEffect, useState } from "react";
+import { useDropzone } from "react-dropzone";
 import { useFormContext } from "react-hook-form";
 
+import { ipfs } from "@/ipfs/utils";
+
 import { Icon } from "../icon";
 import { Caption } from "../typography";
+import { FileUploadProps } from "./types";
 
-interface FileUploadProps {
-  name: string;
-  id?: string;
-  errorMessage?: string;
-  disabled?: boolean;
-}
+export const AddFile = async (fileToUpload: any) => {
+  try {
+    const reader = new FileReader();
+    reader.onabort = () => new Error("file reading was aborted");
+    reader.onerror = () => new Error("file reading has failed");
+    reader.onload = () => {
+      const binaryStr = reader.result;
+      if (binaryStr) {
+        ipfs.add(binaryStr).then((file) => {
+          console.log(file); //IPFS RETURN
+        });
+      }
+    };
+    reader.readAsArrayBuffer(fileToUpload);
+  } catch (error) {
+    console.log(error);
+  }
+};
 
 export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
   function FileUpload({ name, errorMessage, disabled }, ref) {
+    const [files, setFiles] = useState([] as any);
     const formContext = useFormContext();
+
     let _errorMessage = errorMessage;
     if (formContext !== null) {
       _errorMessage = formContext.formState.errors[name]?.message;
     }
     const isError = !!_errorMessage;
 
+    useCallback((acceptedFile) => {
+      AddFile(acceptedFile[0]);
+    }, []);
```

Comment:
Is this useCallback function called as it's not assigned to any variables to use?